src/layers/basic.jl

```diff
@@ -428,3 +429,50 @@ function Base.show(io::IO, m::Parallel)
   join(io, m.layers, ", ")
   print(io, ")")
 end
+
+"""
+    Embedding(in => out; init=randn)
+
+A lookup table that stores embeddings of dimension `out` 
+for a vocabulary of size `in`. 
+
+This layers is often used to store word embeddings and retrieve them using indices. 
+The input to the layer can be either a vector of indexes
+or the corresponding [onehot encoding](@ref Flux.OneHotArray). 
+
+# Examples
+
+```jldoctest
+julia> m = Embedding(reshape(-6:45, 2, 26) .+ 0.01f0)
+Embedding(26 => 2)
+
+julia> m(5)  # embedding vector for 5th element
+2-element Vector{Float32}:
+ 2.01
+ 3.01
+
+julia> m([6, 15, 15])  # applied to a batch
+2×3 Matrix{Float32}:
+ 4.01  22.01  22.01
+ 5.01  23.01  23.01
+
+julia> ans == m(Flux.OneHotMatrix([6, 15, 15], 26))
```

Comment:
My fault, but I think the public function here is the lower-case onehotbatch. This could be
```suggestion
julia> ans == m(Flux.onehotbatch([6, 15, 15], 1:26))
```
or else, according to taste,
```suggestion
julia> ans == m(Flux.onehotbatch(['f', 'o', 'o'], 'a':'z'))
```